mediacloud/mediawords/similarweb/similarweb.py

```diff
@@ -0,0 +1,197 @@
+        resp = self.make_get_request(self.describe_url, {'api_key': self.api_key})
+        resp.raise_for_status()
+        data = resp.json()
+        try:
+            dates = data['response']['total_traffic_and_engagement']['countries']['world']
+        except KeyError:
+            raise KeyError("Expected keys 'response', 'total_traffic_and_engagement', 'countries', 'world', "
+                           "got object {} instead.".format(str(data)))
+        self._start_date = dates['start_date']
+        self._end_date = dates['end_date']
+
+    @property
+    def end_date(self):
+        """Look this up at run time"""
+        if self._end_date is None:
+            self._set_max_date_range()
+        return self._end_date
+
+    @property
+    def start_date(self):
+        """Look this up at run time"""
+        if self._start_date is None:
+            self._set_max_date_range()
+        return self._start_date
+
+    def make_params(self, **kwargs):
+        """Create the query parameter dictionary for a SimilarWeb API call
+
+        Parameters:
+        -----------
+        kwargs :
+            Named keyword arguments are added to the request
+
+        Returns:
+        --------
+        Dictionary that may be passed as query parameters to SimilarWeb
+        """
+        query_args = {
+            'start_date': self.start_date,
+            'end_date': self.end_date,
+            'api_key': self.api_key,
+            'main_domain_only': 'false',
+            'granularity': 'monthly',
+        }
+        query_args.update(kwargs)
+        return query_args
+
+    @staticmethod
+    def tidy_url(url):
+        """Cleans up a url just a little, and removes the initial www (or similar)
+
+        Parameters
+        ----------
+        url : str
+            URL to lightly clean. Perhaps to check for an exact match?
+
+        Returns
+        -------
+        str, lightly cleaned url
+        """
+        tidied = url.lower().strip('/ ')
+        return re.sub(r'(www[0-9]*|m)\.', '', tidied)
+
+    @staticmethod
+    def get_domain(url):
+        """Edit URL to make it appropriate for SimilarWeb API
+
+        SimilarWeb only works on the netloc, with no www.
+
+        Parameters
+        ----------
+        url : str
+            URL to turn into a query for SimilarWeb
+
+        Returns
+        -------
+            str: the domain of url
+        """
+        return SimilarWebClient.tidy_url(urlparse(url).netloc)
+
+    def make_get_request(self, url, params):
+        """Makes API call to SimilarWeb.
+
+        This is a wrapper to allow different methods to make the actual HTTP requests
+
+        Parameters
+        ----------
+            url : str
+                URL to make a GET request to
+            params : dict
+                Query parameters to send along as well
+
+        Returns
+        -------
+        requests.Response
+        """
+        return requests.get(url, params=params)
```

Comment:
Can you please use HTTPAdapter to make this retry for up to 10 minutes in the case of network failure?  As is, a network glitch or briefly down api server will just result in some lines in a log file.  It would be great for us to do a better job monitoring our logs, but regardless it would be best just to be resilient here.

We use the following list of codes for determined retries in our user agent code.  I don't see why we shouldn't just make this constant public so that you can use it directly:

https://github.com/berkmancenter/mediacloud/blob/6fd3b144040c14d4d2cfb20290a2646622fdd075/mediacloud/mediawords/util/web/user_agent/__init__.py#L130